src/cmd/initContainer.go

```diff
@@ -0,0 +1,363 @@
+	flags.BoolVar(&initContFlags.monitorHost, "monitor-host", false, `Ensure that key configuration files (hosts, timezone,..) inside the toolbox container are in sync with the host.`)
+	flags.StringVar(&initContFlags.shell, "shell", "", "Create a user inside the toolbox container whose login shell is SHELL.")
+	initContainerCmd.MarkFlagRequired("shell")
+	flags.IntVar(&initContFlags.uid, "uid", 0, "Create a user inside the toolbox container whose numerical user ID is UID.")
+	initContainerCmd.MarkFlagRequired("uid")
+	flags.StringVar(&initContFlags.user, "user", "", "Create a user inside the toolbox container whose login name is USER.")
+	initContainerCmd.MarkFlagRequired("user")
+}
+
+func initContainer(args []string) {
+	if viper.GetString("XDG_RUNTIME_DIR") == "" {
+		logrus.Info("XDG_RUNTIME_DIR is unset")
+
+		viper.Set("XDG_RUNTIME_DIR", fmt.Sprintf("/run/user/%d", initContFlags.uid))
+		logrus.Infof("XDG_RUNTIME_DIR set to %s", viper.GetString("XDG_RUNTIME_DIR"))
+	}
+
+	toolboxRuntimeDirectory := fmt.Sprintf("%s/toolbox", viper.GetString("XDG_RUNTIME_DIR"))
+	containerInitializedStamp := fmt.Sprintf("%s/container-initialized-%d", toolboxRuntimeDirectory, os.Getpid())
+
+	logrus.Info("Creating /run/.toolboxenv")
+
+	_, err := os.Create("/run/.toolboxenv")
+	if err != nil {
+		logrus.Fatal("Failed to create /run/.toolboxenv")
+	}
+
+	if initContFlags.monitorHost {
+		logrus.Info("Monitoring host")
+
+		if utils.PathExists("/run/host/etc") {
+			logrus.Info("Path /run/host/etc exists. Mount binding to that location will happen now.")
+
+			err = mountBind("/run/host/etc/machine-id", "/etc/machine-id", "ro")
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			err = mountBind("/run/host/run/libvirt", "/run/libvirt", "")
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			err = mountBind("/run/host/run/systemd/journal", "/run/systemd/journal", "")
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			if utils.PathExists("/sys/fs/selinux") {
+				err = mountBind("/usr/share/empty", "/sys/fs/selinux", "")
+				if err != nil {
+					logrus.Error(err)
+				}
+			}
+
+			err = mountBind("/run/host/var/lib/flatpak", "/var/lib/flatpak", "ro")
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			err = mountBind("/run/host/var/log/journal", "/var/log/journal", "ro")
+			if err != nil {
+				logrus.Error(err)
+			}
+
+			err = mountBind("/run/host/var/mnt", "/var/mnt", "rslave")
+			if err != nil {
+				logrus.Error(err)
+			}
+		}
+
+		// TODO: flatpak-session-helper cannot be used over dbus as root; some kind of workaround has to be implemented
+		if utils.PathExists("/run/host/monitor") {
+			logrus.Info("Path /run/host/monitor exists. Mount binding to that path will happen now.")
+			localtimeTarget, err := os.Readlink("/etc/localtime")
+			if err != nil || localtimeTarget != "/run/host/monitor/localtime" {
+				err = redirectPath("/run/host/monitor/localtime", "/etc/localtime", false)
```

Comment:
The semantics of source and target seems to have gotten switched. We redirect `/etc/localtime` to `run/host/monitor/localtime`.